test/Microsoft.AspNetCore.Razor.Test/Framework/TestSpanBuilder.cs

```diff
@@ -139,7 +139,27 @@ public static SpanConstructor Markup(this SpanFactory self, params string[] cont
 
         public static SpanConstructor CodeMarkup(this SpanFactory self, params string[] content)
         {
-            return self.Span(SpanKind.Code, content, markup: true).With(new MarkupChunkGenerator());
+            return self
+                .Span(SpanKind.Code, content, markup: true)
+                .AsCodeMarkup();
+        }
+
+        public static SpanConstructor CSharpCodeMarkup(this SpanFactory self, string content)
```

Comment:
Nope, this is the very sad part about the Razor syntax tree. There's 2 different mechanisms to define what an element is:

1. `Span.Kind`, can be `Code` or `Markup`.
2. `Span.Symbols` type (HTML or CSharp). In this case `CSharpCodeMarkup` only differs from `CodeMarkup` in the symbol type.